pkg/agent/config/config.go

```diff
@@ -467,20 +466,20 @@ func get(ctx context.Context, envInfo *cmds.Agent, proxy proxy.Proxy) (*config.N
 	if !nodeConfig.Docker && nodeConfig.ContainerRuntimeEndpoint == "" {
 		switch nodeConfig.AgentConfig.Snapshotter {
 		case "overlayfs":
-			if err := containerd.OverlaySupported(nodeConfig.Containerd.Root); err != nil {
-				return nil, errors.Wrapf(err, "\"overlayfs\" snapshotter cannot be enabled for %q, try using \"fuse-overlayfs\" or \"native\"",
-					nodeConfig.Containerd.Root)
-			}
+			// if err := containerd.OverlaySupported(nodeConfig.Containerd.Root); err != nil {
+			// 	return nil, errors.Wrapf(err, "\"overlayfs\" snapshotter cannot be enabled for %q, try using \"fuse-overlayfs\" or \"native\"",
+			// 		nodeConfig.Containerd.Root)
+			// }
 		case "fuse-overlayfs":
-			if err := containerd.FuseoverlayfsSupported(nodeConfig.Containerd.Root); err != nil {
-				return nil, errors.Wrapf(err, "\"fuse-overlayfs\" snapshotter cannot be enabled for %q, try using \"native\"",
-					nodeConfig.Containerd.Root)
-			}
+			// if err := containerd.FuseoverlayfsSupported(nodeConfig.Containerd.Root); err != nil {
+			// 	return nil, errors.Wrapf(err, "\"fuse-overlayfs\" snapshotter cannot be enabled for %q, try using \"native\"",
+			// 		nodeConfig.Containerd.Root)
+			// }
 		case "stargz":
-			if err := containerd.StargzSupported(nodeConfig.Containerd.Root); err != nil {
-				return nil, errors.Wrapf(err, "\"stargz\" snapshotter cannot be enabled for %q, try using \"overlayfs\" or \"native\"",
-					nodeConfig.Containerd.Root)
-			}
+			// if err := containerd.StargzSupported(nodeConfig.Containerd.Root); err != nil {
+			// 	return nil, errors.Wrapf(err, "\"stargz\" snapshotter cannot be enabled for %q, try using \"overlayfs\" or \"native\"",
+			// 		nodeConfig.Containerd.Root)
+			// }
```

Comment:
I think we need to bring back some stub version of these checks. I know that we got rid of `pkg/containerd` because the build tags weren't being respected by `go mod`, but the `OverlaySupported` `FuseoverlayfsSupported` and `StargzSupported` (and their stub Windows versions) that we lost should be moved to `pkg/agent/config/config_linux.go` and `pkg/agent/config/config_windows.go` rather than commented out and left unchecked.